google-cloud-spanner/src/test/java/com/google/cloud/spanner/GceTestEnvConfig.java

```diff
@@ -41,6 +45,12 @@
   public static final String GCE_CREDENTIALS_FILE = "spanner.gce.config.credentials_file";
   public static final String GCE_STREAM_BROKEN_PROBABILITY =
       "spanner.gce.config.stream_broken_probability";
+  public static final String ATTEMPT_DIRECT_PATH = "spanner.attempt_directpath";
+  public static final String DIRECT_PATH_TEST_SCENARIO = "spanner.directpath_test_scenario";
+
+  // IP address prefixes allocated for DirectPath backends.
+  public static final String DP_IPV6_PREFIX = "2001:4860:8040";
+  public static final String DP_IPV4_PREFIX = "34.126";
```

Comment:
Will this work outside of GCP? i.e. if you run the system tests with the direct path profile from your local machine?

Reply:
Good question. The short answer is no. DirectPath is designed for clients calling from GCE VM directly, so we actually require that the client need to use the default service account provided by VM itself (see [InstantiatingGrpcChannelProvider](https://github.com/googleapis/gax-java/blob/master/gax-grpc/src/main/java/com/google/api/gax/grpc/InstantiatingGrpcChannelProvider.java#L253)). If user uses other type of credentials (e.g. set GOOGLE_APPLICATIONS_CREDENTIALS with a key file and run client from outside of GCP), then the client would just use CFE path instead of DirectPath.

For testing purpose, I defined this `directpath_test_scenario` property to differentiate three cases that could happen when client opts in to use DirectPath: 1) client is using DP over IPv4, 2) client is using DP over IPv6 and 3) client uses CFE instead. And for cases like running client from my local machine, it should fall in to scenario 3)